comm/src/main/scala/coop/rchain/comm/transport/StreamHandler.scala

```diff
@@ -22,6 +22,10 @@ object StreamHandler {
 
   type CircuitBreaker = Long => Boolean
 
+  sealed trait StreamFault
+  final case class StreamError(message: String)      extends StreamFault
```

Comment:
please change those names, they are meaning less

Reply:
@rabbitonweb you didn't propose any better names I should use instead ;)

Reply:
A stream error is when the other side did something wrong. It's a user-side mistake. On the other hand, a stream exception is an unexpected error. So it's a server-side problem. Both errors together form a disjoint stream fault algebra.

Reply:
I agree that the error names make little sense. You can probably take `CommError`'s approach here and split user-level errors (`StreamError`) into different classes (e.g. `WrongNetworkId`, `CircuitWasBroken`). Also, it seems to me that server-side errors mostly come from unexpected IO errors so you could introduce `StreamUnexpectedIOError` or something similar.